pkg/prometheus/promcfg.go

```diff
@@ -4104,6 +4114,24 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 	return cfg, nil
 }
 
+func (cg *ConfigGenerator) generateOTLPConfig(_ *assets.StoreBuilder) yaml.MapItem {
```

Comment:
wdyt about receive the otlp config as a parameters since on the place where we're invoking generateOTLPConfig we're already checking if the OTLP field from the Common Fields is != nil.

Reply:
we don't even need an otlp config parameter since it's coming from the Prometheus spec for which the config generator has a copy.